### ISSUE TITLE
golangci-lint is failing with go version 1.20

### ISSUE THREAD

**Issue by Tulsishah** [golangci/golangci-lint#3803]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

I am running [lint tests](https://github.com/Tulsishah/gcsfuse-tulsishah7/blob/updating_go_version_119_to_120/.github/workflows/ci.yml#L41) on my repo.
with go version 1.20 

but it is throwing this error 

 Running [/home/runner/golangci-lint-1.52.2-linux-amd64/golangci-lint run --out-format=github-actions --new-from-patch=/tmp/tmp-1700-XWwsqXhSP6L9/pull.patch --new=false --new-from-rev= -E=goimports] in [] ...
  level=warning msg="Failed to discover go env: failed to run 'go env': exit status 2"
  level=error msg="Running error: context loading failed: failed to load packages: failed to load with go/packages: err: exit status 1: stderr: template: main:1: function \"context\" not defined\n"
  
  Error: golangci-lint exit with code 3
  Ran golangci-lint in 124ms

However for go version 1.19 it is working successfully. Can I know the reason or possible solution ?

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci/golangci-lint-action@032fa5c5e48499f06cf9d32c02149bfac1284239
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
# paste output here
```
-

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```
go version 1.20

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```
-
</details>

### Code example or link to a public repository

<details>

```go
// add your code here
```
https://github.com/Tulsishah/gcsfuse-tulsishah7/blob/master/.github/workflows/ci.yml

</details>

**Comment by bombsimon** [comment]
You're using 

```yaml
- name: golangci-lint
  uses: golangci/golangci-lint-action@032fa5c5e48499f06cf9d32c02149bfac1284239
```

`032fa5c` is an old commit from August 2022, try changing that to `golangci/golangci-lint-action@v3`.

**Comment by Tulsishah** [comment]
> You're using
> 
> ```yaml
> - name: golangci-lint
>   uses: golangci/golangci-lint-action@032fa5c5e48499f06cf9d32c02149bfac1284239
> ```
> 
> `032fa5c` is an old commit from August 2022, try changing that to `golangci/golangci-lint-action@v3`.

I did it same, giving same error
